File: app/src/main/java/com/example/getfood/ui/loginregister/LoginActivityMvpPresenter.java
package com.example.getfood.ui.loginregister;

import android.content.Intent;

import com.example.getfood.ui.base.BaseMvpPresenter;

public interface LoginActivityMvpPresenter<V extends LoginActivityMvpView> extends BaseMvpPresenter<V> {
    void bindDynamicLinkEmailVerification(Intent intent);

    boolean isUserLoggedIn();

    void signOutUser();
}

File: app/src/main/java/com/example/getfood/ui/loginregister/LoginActivityMvpPresenter.kt
package com.example.getfood.ui.loginregister

import android.content.Intent
import com.example.getfood.ui.base.BaseMvpPresenter

open interface LoginActivityMvpPresenter<V : LoginActivityMvpView?> : BaseMvpPresenter<V> {
    fun bindDynamicLinkEmailVerification(intent: Intent?)
    val isUserLoggedIn: Boolean
    fun signOutUser()
    fun updateToken(token: String?)
}